sender.go

```diff
@@ -144,6 +169,63 @@ func WithFileNameLimit(limit int) LineSenderOption {
 	}
 }
 
+func WithGraceTimeout(timeout time.Duration) LineSenderOption {
```

Comment:
Ok cool! I was going off the document (https://docs.google.com/document/d/1EQZjYrJO_Ll8RrSIZpbr15Ul3N-xAtTxvrunI-TpOR8/edit#heading=h.lrrfjiq35q9t) but I guess things have changed since then? HTTP timeout makes much more sense to me

Reply:
> but I guess things have changed since then?

Probably. @jerrinot do you happen to know why Java HTTP sender has HTTP timeout, but not those grace settings?

Reply:
I discussed this with @amunra and I'm going to add this to Java client too

Reply:
OK, we'll have to keep using these new settings then. Resolving this one.